internal/provider/kubernetes/controller.go

```diff
@@ -196,7 +207,41 @@ func newGatewayAPIController(mgr manager.Manager, cfg *config.Server, su status.
 	return nil
 }
 
+func (r *gatewayAPIReconciler) enqueueManagedClass(obj client.Object) []reconcile.Request {
+	ep, ok := obj.(*egcfgv1a1.EnvoyProxy)
+	if !ok {
+		panic(fmt.Sprintf("unsupported object type %T", obj))
+	}
+
+	// The EnvoyProxy must be in the same namespace as EG.
```

Comment:
it's a bit confusing here
`EnvoyProxy` is namespaced resource, but required to same namespace as EG, then namespace is ignored after enqueued.